client/src/ThingsExplorer/ItemsProvider.ts

```diff
@@ -1,23 +1,53 @@
 import {
     SnippetString,
-    window
+    window,
+    workspace
 } from 'vscode'
 
 import * as _ from 'lodash'
-import * as s from 'underscore.string'
 import * as AsciiTable from 'ascii-table'
 
 import { Thing } from './Thing'
 import { Channel } from './Channel'
 
+/** generate an Item name from a Thing label by using the configured casing */
+function generateItemName(label: string) : string {
+    const config = workspace.getConfiguration('openhab')
+    switch(config.get('itemCasing')) {
+        case 'snake':
+            // upper snake case, 'Guest Room Light' -> 'Guest_Room_Light'
+            return _.startCase(label).replace(/ /g, "_");
+        default:
+            // camel case, 'Guest Room Light' -> 'GuestRoomLight'
+            return _.startCase(label).replace(/ /g, "");
+    }

```

Comment:
DRY
```
    let casing = '';
    switch(config.get('itemCasing')) {
        case 'snake':
            // upper snake case, 'Guest Room Light' -> 'Guest_Room_Light'
            casing = '_';
        default:
            // camel case, 'Guest Room Light' -> 'GuestRoomLight'
            casing = '';
    }
    return _.startCase(label).replace(/ /g, casing);
```

Reply:
@kubawolanin 

while it's true that this could be considered a DRY case, I'd propose to leave it as it. If additional casing options are to come, or if one of those casings would not be that similar to the other, it would be a tough one to *reverse* that refactoring. It creates a dependency between them, that is not necessary. Having it the way it is right now on the other hand makes it clear how the different *casings* are build, and allow for additions or to rewrite them independently. 

What do you say?

Reply:
Hmpf.
Seriously nothing that should block this PR any longer...

I am perfectly fine with leaving it this way now.